core/src/main/java/org/wso2/testgrid/tinkerer/UnixClient.java

```diff
@@ -0,0 +1,157 @@
+ * under the License.
+ */
+package org.wso2.testgrid.tinkerer;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.io.FileUtils;
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.client5.http.fluent.Response;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.http.HttpHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.Agent;
+import org.wso2.testgrid.common.DeploymentCreationResult;
+import org.wso2.testgrid.common.Host;
+import org.wso2.testgrid.common.TestGridConstants;
+import org.wso2.testgrid.common.TestPlan;
+import org.wso2.testgrid.common.util.StringUtil;
+import org.wso2.testgrid.common.util.TestGridUtil;
+import org.wso2.testgrid.tinkerer.exception.TinkererOperationException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is the {@link TinkererClient} implementation for agents that are running in
+ * UNIX operating systems.
+ *
+ * @since 1.0.0
+ *
+ */
+public class UnixClient extends TinkererClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(UnixClient.class);
+    private static final String SCENARIO_LOG_LOCATION = "/repository/logs/";
+    private static final String INTEGRATION_LOG_LOCATION = "/logs";
+
+    @Override
+    public void downloadLogs(DeploymentCreationResult deploymentCreationResult, TestPlan testPlan)
+            throws TinkererOperationException {
+
+        String testType = testPlan.getScenarioConfig().getTestType();
+        logger.info("TestType is : " + testType);
+
+        try {
+            for (Agent agent : deploymentCreationResult.getAgents()) {
+                String productBasePath = "";
+                String logLocation = "";
+                String operationPath = this.getTinkererBase() + "test-plan/" + agent.getTestPlanId()
+                        + "/agent/" + agent.getInstanceName() + "/operation";
+                if (TestGridConstants.TEST_TYPE_FUNCTIONAL.equals(testType)) {
+                    //get the carbon.home from the instance where the product is running
+                    Response execute = Request.Post(operationPath).setHeader
+                            (HttpHeaders.CONTENT_TYPE, "application/json")
+                            .setHeader(HttpHeaders.AUTHORIZATION, "Basic " + Base64.getEncoder()
+                                    .encodeToString(StringUtil.concatStrings(this.getTinkererUserName(),
+                                            ":", this.getTinkererPassword()).getBytes(Charset.defaultCharset())))
+                            .bodyString("{\"code\":\"SHELL\",\"request\":\"ps -ef |grep 'carbon.home'\"}",
```

Comment:
Would it be better if we create a JSON object and add those elements and then convert it back to String? :)